cmd/aro/gateway.go

```diff
@@ -44,38 +38,25 @@ func gateway(ctx context.Context, log *logrus.Entry) error {
 	if err := env.ValidateVars(envDatabaseAccountName); err != nil {
 		return err
 	}
-	dbc, err := database.NewDatabaseClient(log.WithField("component", "database"), _env, nil, m, nil, os.Getenv(envDatabaseAccountName))
+
+	msiToken, err := _env.NewMSITokenCredential()
 	if err != nil {
 		return err
 	}
+	logrusEntry := log.WithField("component", "database")
 
-	// Access token GET request needs to be:
-	// http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=$AZURE_DBTOKEN_CLIENT_ID
-	//
-	// In this context, the "resource" parameter is passed to azidentity as a
-	// "scope" argument even though a scope normally consists of an endpoint URL.
-	scope := os.Getenv("AZURE_DBTOKEN_CLIENT_ID")
-	msiRefresherAuthorizer, err := _env.NewMSIAuthorizer(scope)
+	dbName, err := DBName(_env.IsLocalDevelopmentMode())
 	if err != nil {
 		return err
 	}
 
-	// TODO: refactor this poor man's feature flag
-	insecureSkipVerify := _env.IsLocalDevelopmentMode()
-	for _, feature := range strings.Split(os.Getenv("GATEWAY_FEATURES"), ",") {
-		if feature == "InsecureSkipVerifyDBTokenCertificate" {
-			insecureSkipVerify = true
-			break
-		}
-	}
-
-	url, err := getURL(_env.IsLocalDevelopmentMode())
+	dbAccountName := os.Getenv(envDatabaseAccountName)
+	scope := []string{fmt.Sprintf("https://%s.%s", dbAccountName, _env.Environment().CosmosDBDNSSuffixScope)}
```

Comment:
There's no `CosmosDBDNSSuffixScope` variable for any environment. Is `CosmosDBDNSSuffix` needed here?This can be probably the reason of VMSS bootstrap getting timed out.